### DISCUSSION TITLE
Renovate for specific upgrades and does it work with ComponentGovernance alert

### DISCUSSION THREAD

**Discussion by dipeshkalyani** [renovatebot/renovate#17860] [Request Help] [answered]
Hi Team,

I have just started reading about Renovate and its quite powerful.
To suit to my existing workflow, I had two queries:
1. Can I somehow invoke renovate to just upgrade say, PackageA to a higher version?
2. Can I configure Renovate to be invoked for ComponentGovernance alerts in ADO?
3. How can I make it work for cases where packages versions are specified in a config file (say corext.config) and not in any csproj? What I read, it seems it doesn't seem to support it.

**Comment by Gabriel-Ladzaretti** [comment, marked as answer]
Hello @dipeshkalyani and welcome 😀


1) Can be achieved using [packageRules](https://docs.renovatebot.com/configuration-options/#packagerules)
    - First: [disable ](https://docs.renovatebot.com/configuration-options/#enabled)all other package updates using [matchPackagePatterns](https://docs.renovatebot.com/configuration-options/#matchpackagepatterns) matcher 
    - Second: [enable ](https://docs.renovatebot.com/configuration-options/#enabled)that one package with [matchPackageNames](https://docs.renovatebot.com/configuration-options/#matchpackagenames) matcher 


e.g.

```json
{
  "packageRules": [
    {
      "matchPackagePatterns": [".*"],
      "enabled": false
    },
    {
      "matchPackageNames": ["packageName"],
      "enabled": true
    }
  ]
}
```


3) This can probably be acheived using a [regexManager](https://docs.renovatebot.com/configuration-options/#regexmanagers)

**Comment by rarkins** [comment]
2. No, how would that work?

**Reply by dipeshkalyani** [reply]
Thanks @Gabriel-Ladzaretti.
@rarkins : For (2), one workflow I can think of is querying the alerts and then using (1)'s solution to create PRs.

**Reply by rarkins** [reply]
Unfortunately it's a bit too much like this:

![image](https://user-images.githubusercontent.com/6311784/191172890-32531b8a-a1dd-4313-9ec6-d50407f7b6cb.png)
